### PR TITLE
More specific Airframe type support for better Airframe Reference

### PR DESCRIPTION
**Pull request by junwoo091400**
**Describe problem solved by this pull request**
As discussed in the Actuators Document Improvement PR in User Guide: https://github.com/PX4/PX4-user_guide/pull/1858#discussion_r876554728, **the current airframe reference document doesn't separate out the airframes accurately.**

image

For example, the 'VTOL Tiltrotor' [Airframe Reference](http://docs.px4.io/master/en/airframes/airframe_reference.html#vtol-tiltrotor) has a **6-rotor VTOL diagram that doesn't match with the E-Flite Convergence, that is currently under the same "type" in the doc.**

**Describe your solution**
1. Add more @type categories to the [Airframe files](https://github.com/PX4/PX4-Autopilot/tree/master/ROMFS/px4fmu_common/init.d/airframes) to specify the Airframes that is currently in the wrong 'type'
2. Add the SVG diagrams in the User Doc for those specific @type categories, which will be a separate PR

**TODOs**
- [x] Cleanup Airframe Source Parser to clarify the 'airframe' object (which was named 'param' before)
- [ ] Rename E-Flite Convergence Airframe's @type to a more specific type, decoupled from a generic "VTOL Tiltrotor"

**Test data / coverage**
Tested functionality of the `px_process_airframes.py` script by: `make airframe_metadata` command

**Additional context**
🤞  If there are any other airframes that are not true to the airframe type diagram in the [Airframe Reference](http://docs.px4.io/master/en/airframes/airframe_reference.html), please let me know so that I can add the appropriate @type attribute to those airframes!

### CODE REVIEW

**Comment by junwoo091400** [comment]
@sfuhrer Any idea for a better @type for the Convergence airframe? https://github.com/PX4/PX4-Autopilot/blob/master/ROMFS/px4fmu_common/init.d/airframes/13012_convergence#L5

**Comment by sfuhrer** [comment]
> @sfuhrer Any idea for a better @type for the Convergence airframe? https://github.com/PX4/PX4-Autopilot/blob/master/ROMFS/px4fmu_common/init.d/airframes/13012_convergence#L5


Not really, currently we don't have a way to indicate the number of motors of a tiltrotor.

Given that we want to purge most of the airframes once we have control allocation in place (see https://github.com/PX4/PX4-Autopilot/pull/19596#issuecomment-1119658490), I wonder how much effort it is worth to spend on this front. The idea is that we only keep a very small selection of pre-configured airframes and otherwise let users define their own airframe with the new allocation UI in QGC.

**Comment by sfuhrer** [comment]
> @sfuhrer Any idea for a better @type for the Convergence airframe? https://github.com/PX4/PX4-Autopilot/blob/master/ROMFS/px4fmu_common/init.d/airframes/13012_convergence#L5


Not really, currently we don't have a way to indicate the number of motors of a tiltrotor.

Given that we want to purge most of the airframes once we have control allocation in place (see https://github.com/PX4/PX4-Autopilot/pull/19596#issuecomment-1119658490), I wonder how much effort it is worth to spend on this front. The idea is that we only keep a very small selection of pre-configured airframes and otherwise let users define their own airframe with the new allocation UI in QGC.

**Comment by junwoo091400** [comment]
This would be a useful refactor in preparation for the upcoming Frame component metadata generation in PX4 side.

It is just renaming / making the script more human readable, so no functional changes.

Could we get this in? @bkueng

File: Tools/px4airframes/srcparser.py [hunk 1]
```diff
@@ -2,27 +2,30 @@
...
         """
         Add parameter to the group
         """
-        self.params.append(param)
+        self.airframes.append(airframe)
 
     def GetName(self):
         """
-        Get parameter group name
+        Get parameter group type
```
Comment (bkueng) [inline Tools/px4airframes/srcparser.py:26]: Can you fix the whole comment? parameter -> airframe
And why does `GetName` return `type`?

Comment (junwoo091400) [reply Tools/px4airframes/srcparser.py:26]: Sure! That's also what I wondered back then I think :laughing: 

This script in general has bad naming conventions / implicit, but I will go over them again

Comment (junwoo091400) [reply Tools/px4airframes/srcparser.py]: Fixed!

**Comment by junwoo091400** [comment]
Had rebase error, now properly fixed!

**Review by bkueng — APPROVED** [review]

**Pull request merged**